app/src/main/java/com/eveningoutpost/dexdrip/AddCalibration.java

```diff
@@ -195,16 +203,31 @@ public void onClick(final View v) {
                             final double calValue = JoH.tolerantParseDouble(string_value);
 
                             if (!Home.get_follower()) {
-                                Calibration calibration = Calibration.create(calValue, getApplicationContext());
-                                if (calibration != null) {
-                                    UndoRedo.addUndoCalibration(calibration.uuid);
-                                    //startWatchUpdaterService(v.getContext(), WatchUpdaterService.ACTION_SYNC_CALIBRATION, TAG);
-                                    //Ob1G5StateMachine.addCalibration((int)calibration.bg, calibration.timestamp);
-                                    NativeCalibrationPipe.addCalibration((int)calibration.bg, calibration.timestamp);
+                                if (FirmwareCapability.isTransmitterRawIncapable(getTransmitterID())) { // Firefly only
```

Comment:
Can you explain the changes in `AddCalibration` is this because creating a calibration results in a null if the sensor sanity tests fail on the raw value? So you're creating a blood test entry instead?

Reply:
My first step was to add the add calibration option in the menu only for a Firefly.  This was done by the added fork to NavDrawerBuilder.java.
After I did that, I saw that the calibrations still did not go through for a Firefly.
I realized that Calibration.java had a fork that checked for raw sanity and just rejected a Firefly.

My first attempt to fix that was to modify Calibration.java.  It became too complex.  So, I decided to take a different approach.  I created a fork in AddCalibration.java.  This fork side is only taken for a Firefly.  A G5 or an old G6 just goes through as before.

The Firefly side of the fork, in AddCalibration,java needs to do everything that is done in AddCalibration.java as well as what is done in Calibration.java that is not executed for a Firefly.  Those are creating a blood glucose reading point on the plot as well as sending the calibration to the transmitter.

If there is a more graceful way of accomplishing this, please tell me.  I am so sorry if this is messed up.  
I can also add more comments to the code to explain all of this better if you like.

Thanks for looking.